### ISSUE TITLE
Lots of Validations/Verifications are needed

### ISSUE THREAD

**Issue by G0maa** [G0maa/my-school-server#34]
* Taken from #18, as it's an important requirement/issue.
* Validations (lots of them)
* `POST` request for all entities should verify some things, e.g.: 
	* tl;dr, easier implementation for now: can't `POST` `ActiveSubject` without specifying `StudyClass` & `Subject` & `Teacher`. ❌
		* a `Student` can't be in a `studyClass` that has a different `educationType` or `studyYear`.
		* Currently a `Teacher` can teach any `Subject` (in table `ActiveSubjects`) which isn't "real".
		* Similar to validaitons in `Student`, adding a `Subject` to `ActiveSubjects` should verify that `studyClass` in the relationship has the correct `studyYear` & `educationType`
	* To-be-edited, lots of stuff left.
-   `PUT` request for all entities should have some validations, because they may result in an invalid state 
	-  tl;dr easier implementation for now: **just remove that particular entitiy from its relations**. ❌
	- Even more simple: Ask `Admin` to go and delete `Student` from said `StudyClass`
		- Similarly with `Subject` & `ActiveSubject`
		- `Teacher` isn't affected by this.
	- To-be-edited, lots of stuff left.
